### PR TITLE
Caterva_meta_print

### PR DESCRIPTION
**Pull request by oscargm98**
Add new function "caterva_meta_print" to Caterva

### CODE REVIEW

**Comment by aleixalcacer** [comment]
I think we should not introduce this new feature (as is) for the following reasons:

1. The API `caterva_meta_*` is meant for all existing metalayers. However, `caterva_meta_print` will only work with the Caterva metalayer.
2. The API `caterva_meta_*` requires a pointer to a Caterva array. However, `caterva_meta_print` will only work with persistent arrays.

For this reasons, I'm going to propose an alternative:

```
int caterva_print_meta_info(caterva_ctx_t *ctx, caterva_array_t *array) {
    ...
}
```

Thus, if you want to print the info of a persistent array, you just need to open the array before calling this function.

What dou you think @FrancescAlted ?

Apart from this, could you (@oscargm98) give an example of what the output would look like?

**Comment by FrancescAlted** [comment]
Yes, the idea was to have a utility for printing the metalayer info for existing Caterva frames.  But I agree that @aleixalcacer proposal adds more flexibility, so +1 to it.  Also, I am not a huge fan of long names, so I'd propose `caterva_print_meta()` instead.

**Comment by oscargm98** [comment]
Requested changes done! You can see the result running example_frame_generator, which will have even more examples when its PR is merged

File: examples/example_print_meta.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    caterva_array_t *arr;
+    CATERVA_ERROR(caterva_open(ctx, urlpath, &arr));
+    caterva_print_meta(arr);
+    caterva_free(ctx, &arr);
+    caterva_ctx_free(&ctx);
+
+    return 0;
+}
+
+int main() {
```
Comment (FrancescAlted) [inline examples/example_print_meta.c]: You may want to add a caterva file as a parameter so as to print meta info for it.

**Review by FrancescAlted — CHANGES_REQUESTED** [review]

File: examples/example_print_meta.c (hunk 1 above)
Comment (oscargm98) [reply examples/example_print_meta.c]: Done!

File: examples/example_print_meta.c [hunk 2]
```diff
@@ -6,7 +6,18 @@
  * LICENSE file in the root directory of this source tree) and the GPLv2 (found
  * in the COPYING file in the root directory of this source tree).
  * You may select, at your option, one of the above-listed licenses.
- */
+ * Test program demonstrating use of the Blosc codec from C code.
+ * To compile this program:
+ *
+ * $ gcc -O example_print_meta.c -o example_print_meta -lblosc2
+ * $ <urlpath>
```
Comment (FrancescAlted) [inline examples/example_print_meta.c:13]: You rather mean `$ ./example_print_meta <urlpath1> <urlpath2>...` ?

**Review by FrancescAlted — APPROVED** [review]
LGTM.  Just fix my comment, and you can merge it.

File: examples/example_print_meta.c (hunk 2 above)
Comment (FrancescAlted) [reply examples/example_print_meta.c:13]: Fixed in afc5a79

**Pull request merged**